lib/Dialect/Torch/Transforms/AdjustCallingConventions.cpp

```diff
@@ -191,7 +191,10 @@ static bool isValidNonContainerResultType(Type resultType) {
          resultType.isa<Torch::FloatType>() ||
          resultType.isa<Torch::IntType>() ||
          resultType.isa<Torch::BoolType>() ||
-         resultType.isa<Torch::NoneType>();
+         resultType.isa<Torch::NoneType>() ||
+         (resultType.isa<Torch::ListType>() && cast<Torch::ListType>(resultType)
```

Comment:
We shouldn't be doing this. This function was created with the goal of preventing something like a `ListType` return to reach the backend contract. This would lead to invalid IR being generated.

Reply:
That's fine but if you want to enable user-provided shape and dtype functions in the same parent module then there needs to be special casing for them. The alternative, is to provide some mechanism for passing handles to a `ModuleOp` all the way down into `wrapWithCalculateOpIfLibraryFunctionAvailable`.

Reply:
User provided shape and dtype functions will be handled exactly the same way that current shape and dtype functions are handled. The plan is to load them from a `.mlir` file when the shape or dtype pass is taking place, then remove them once the pass is done. This is all outlined in the RFC: https://github.com/llvm/torch-mlir/issues/1462. There shouldn't be any coupling between the shape and dtype pipelines and the rest of the passes in torch-mlir.